pkg/network/ebpf/c/prebuilt/offset-guess.c

```diff
@@ -10,6 +10,38 @@
 #include <uapi/linux/tcp.h>
 #include <uapi/linux/ip.h>
 
+#define sizeof_member(T, m) sizeof(((T*)0)->m)
+
+static const __u8 SIZEOF_SADDR = sizeof_member(tracer_status_t, saddr);
+static const __u8 SIZEOF_DADDR = sizeof_member(tracer_status_t, daddr);
+static const __u8 SIZEOF_FAMILY = sizeof_member(tracer_status_t, family);
+static const __u8 SIZEOF_SPORT = sizeof_member(tracer_status_t, sport);
+static const __u8 SIZEOF_DPORT = sizeof_member(tracer_status_t, dport);
+static const __u8 SIZEOF_NETNS = sizeof((struct net*)0);
+static const __u8 SIZEOF_NETNS_INO = sizeof_member(tracer_status_t, netns);
+static const __u8 SIZEOF_RTT = sizeof_member(tracer_status_t, rtt);
+static const __u8 SIZEOF_RTT_VAR = sizeof_member(tracer_status_t, rtt_var);
+static const __u8 SIZEOF_DADDR_IPV6 = sizeof_member(tracer_status_t, daddr_ipv6) / 4;
+static const __u8 SIZEOF_SADDR_FL4 = sizeof_member(tracer_status_t, saddr_fl4);
+static const __u8 SIZEOF_DADDR_FL4 = sizeof_member(tracer_status_t, daddr_fl4);
+static const __u8 SIZEOF_SPORT_FL4 = sizeof_member(tracer_status_t, sport_fl4);
+static const __u8 SIZEOF_DPORT_FL4 = sizeof_member(tracer_status_t, dport_fl4);
+static const __u8 SIZEOF_SADDR_FL6 = sizeof_member(tracer_status_t, saddr_fl6) / 4;
+static const __u8 SIZEOF_DADDR_FL6 = sizeof_member(tracer_status_t, daddr_fl6) / 4;
+static const __u8 SIZEOF_SPORT_FL6 = sizeof_member(tracer_status_t, sport_fl6);
+static const __u8 SIZEOF_DPORT_FL6 = sizeof_member(tracer_status_t, dport_fl6);
+static const __u8 SIZEOF_SOCKET_SK = sizeof(uintptr_t);
+static const __u8 SIZEOF_SK_BUFF_SOCK = sizeof(uintptr_t);
+static const __u8 SIZEOF_SK_BUFF_TRANSPORT_HEADER = sizeof_member(tracer_status_t, transport_header);
+static const __u8 SIZEOF_SK_BUFF_HEAD = sizeof(uintptr_t);
+
+static __always_inline u64 aligned_offset(void *p, u64 offset, uintptr_t size) {
+    u64 _p = (u64)p;
+    _p += offset;
+    _p = _p + size - 1 - (_p + size - 1) % size;
```

Comment:
That would yield the floor value, we want the ceiling.

Reply:
Could you do `_p - (_p % size) + size`?

Reply:
DMd and now I understand, using the ceiling means that the offsetguesser will automatically advance by `size`;  but the candidate offset is aligned to `size` .

That means the candidate offset starts at at 0 (from go), then gets advanced to (1) by go, then gets advanced to 4 the next time aligned_offset is called.   

I think a comment to this effect would be helpful (or writing this with an if statement instead of this wild math) to make it clearer what's happening.